app.js

```diff
@@ -69,7 +69,8 @@ document.querySelector( '#validate-form' ).addEventListener( 'click', () => {
 function initializeForm() {
     const formEl = document.querySelector( 'form.or' );
     form = new Form( formEl, {
-        modelStr: modelStr
+        modelStr: modelStr,
+        instanceStr: new DOMParser().parseFromString( modelStr, 'text/xml' ).querySelector( 'model > instance' ).innerHTML
```

Comment:
I don't understand what your purpose is here... The bug is related to the record, that you're editing. So you have to actually use the complete record that is creating the problem.

Reply:
If I do not pass this value here to `instanceStr` then it stays undefined and then `mergeXml` is never called [here](https://github.com/enketo/enketo-core/blob/master/src/js/form-model.js#L186). Unless you mean I'm assigning the wrong value to it, it seems important to me that `instanceStr` is assigned a value. Please let me know what you think about this. Thanks

Reply:
It should be the **actual record** that is causing the problem. How are you otherwise going to reproduce the problem? 

So like: 

```js
instanceStr: `<data id="test_school_form">
                    <start_time/>
                    <end_time/>
                    <device_id/>
                    <group1>
                        <repeat1>
                            <position_repeat1>1</position_repeat1>
                            <numb_buid99>1</numb_buid99>
                            <building>bulding_type_list7</building>
                            <repeat11>
                                <position_repeat11>1</position_repeat11>
                                <total_numb_corridors>1</total_numb_corridors>
                                <repeat111>
                                    <position_repeat111>1</position_repeat111>
                                    <total_numb_rooms>1</total_numb_rooms>
                                    <room_open_stat>no</room_open_stat>
                                </repeat111>
                                <repeat111>
                                    <position_repeat111>2</position_repeat111>
                                    <total_numb_rooms>1</total_numb_rooms>
                                    <room_open_stat>no</room_open_stat>
                                </repeat111>
                                <total_number_room>1</total_number_room>
                            </repeat11>
                            <repeat11>
                                <position_repeat11>2</position_repeat11>
                                <total_numb_corridors>1</total_numb_corridors>
                                <repeat111>
                                    <position_repeat111>1</position_repeat111>
                                    <total_numb_rooms>1</total_numb_rooms>
                                    <room_open_stat>no</room_open_stat>
                                </repeat111>
                                <total_number_room>6</total_number_room>
                            </repeat11>
                            <total_number_cord>1</total_number_cord>
                        </repeat1>
                        <repeat1>
                            <position_repeat1>2</position_repeat1>
                            <numb_buid99>1</numb_buid99>
                            <building>bulding_type_list4</building>
                            <repeat11>
                                <position_repeat11>1</position_repeat11>
                                <total_numb_corridors>1</total_numb_corridors>
                                <repeat111>
                                    <position_repeat111>1</position_repeat111>
                                    <total_numb_rooms>1</total_numb_rooms>
                                    <room_open_stat>no</room_open_stat>
                                </repeat111>
                                <total_number_room>2</total_number_room>
                            </repeat11>
                            <total_number_cord>2</total_number_cord>
                        </repeat1>
                        <repeat1>
                            <position_repeat1>3</position_repeat1>
                            <numb_buid99>1</numb_buid99>
                            <building>bulding_type_list7</building>
                            <repeat11>
                                <position_repeat11>1</position_repeat11>
                                <total_numb_corridors>1</total_numb_corridors>
                                <repeat111>
                                    <position_repeat111>1</position_repeat111>
                                    <total_numb_rooms>1</total_numb_rooms>
                                    <room_open_stat>no</room_open_stat>
                                </repeat111>
                                <total_number_room>3</total_number_room>
                            </repeat11>
                            <total_number_cord>3</total_number_cord>
                        </repeat1>
                        <repeat1>
                            <position_repeat1>4</position_repeat1>
                            <numb_buid99>1</numb_buid99>
                            <building>bulding_type_list4</building>
                            <repeat11>
                                <position_repeat11>1</position_repeat11>
                                <total_numb_corridors>1</total_numb_corridors>
                                <repeat111>
                                    <position_repeat111>1</position_repeat111>
                                    <total_numb_rooms>1</total_numb_rooms>
                                    <room_open_stat>no</room_open_stat>
                                </repeat111>
                                <total_number_room>4</total_number_room>
                            </repeat11>
                            <total_number_cord>4</total_number_cord>
                        </repeat1>
                    </group1>
                    <total_numb_b/>
                    <meta>
                        <instanceID/>
                    </meta>
                </data>`,

```

Reply:
Note that app.js is just a minimal template for developers to be able to run enketo-core (which is a library, not an app) locally without enketo-express.

So this edit is just for during development allowing you to reproduce and fix the bug.